pkg/controller/cluster/controller.go

```diff
@@ -120,16 +124,32 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, nil
 	}
 
-	current := cr.Spec.ForProvider.DeepCopy()
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	currentStatus := cr.Status.AtProvider.DeepCopy()
 	lateInitializeCluster(&cr.Spec.ForProvider, observedCluster)
 
-	cr.Status.AtProvider = generateClusterObservation(observedCluster)
+	cr.Status.AtProvider = v1alpha1.ClusterObservation{}
+
+	// check if kubeconfig secret has been updated
+	kubeconfigSecretUpToDate := true
+	if cr.Spec.ForProvider.Config.KubeconfigSecretRef != nil {
+		kubeconfigSecretResourceVersion, err := e.getResourceVersion(ctx, cr.Spec.ForProvider.Config.KubeconfigSecretRef)
+		if err != nil {
+			return managed.ExternalObservation{}, err
+		}
+		kubeconfigSecretUpToDate = cr.Status.AtProvider.Kubeconfig != nil &&
+			(kubeconfigSecretResourceVersion == currentStatus.Kubeconfig.Secret.ResourceVersion)
+		cr.Status.AtProvider.Kubeconfig = &v1alpha1.KubeconfigObservation{
+			Secret: v1alpha1.SecretObservation{ResourceVersion: kubeconfigSecretResourceVersion},
+		}
+	}
```

Comment:
This should be in a separate function, i.e. `isKubeConfigSecretUpToDate`

Reply:
I was able to compare using `cmp.Equal`